src/input.rs

```diff
@@ -0,0 +1,89 @@
+use crate::model::*;
+use egg::*;
+use std::collections::HashMap;
+
+
+/// Struct for converting a model specified using our Rust interface to RecExpr
+///
+/// The RecExpr is growed on the fly when member functions are called. Uses a 
+/// Hashmap to store the map of scalar nodes to their index into the RexExpr to
+/// avoid replication.
+#[derive(Default)]
+pub struct GraphConverter {
+    rec_expr: RecExpr<Mdl>,
+    scalar_map: HashMap<i32, Id>,
+}
+
+
+impl GraphConverter {
```

Comment:
Add a comment here saying that the APIs of this class are (intended to be) the same as TASO's so that we can easily constructing TASO graphs using this class. Am I correct?

Reply:
Yes that's right! Comment added